routing_common/transit_types.hpp

```diff
@@ -104,20 +104,16 @@ class Gate
   m2::PointD const & GetPoint() const { return m_point; }
 
   DECLARE_VISITOR_AND_DEBUG_PRINT(Gate, visitor(m_featureId, "osm_id"),
-                                  visitor(m_entrance, "entrance"),
-                                  visitor(m_exit, "exit"), visitor(m_weight, "weight"),
-                                  visitor(m_stopIds, "stop_ids"), visitor(m_point, "point"))
+                                  visitor(m_pedestrianFeatureIds, "" /* name */),
```

Comment:
We cann't. Because feature ids are kept at the vector (|m_pedestrianFeatureIds|). As I notice before, they are generated on transit section generation step.

Reply:
So why do you use "osm_id" for visitors? Looks like an error and raises a lot of questions.

Reply:
Are you asking about |m_featureId| field in Gate class?
|osm_id| is a field name at json of transit graph. It has the name because it contains osm id of some feature. The field should be converted to feature id before writing it to mwm and kept at |m_featureId|.

What can we do about it?
* At DeserializerFromJson visitor reading json field name (osm id) and converting its value to feature id, and keeping it at |m_featureId| field;
* Adding field |m_osm_id| to Gate and leaving |m_featureId| field in Gate. In DeserializerFromJson filling |m_osm_id| field of Gate. After that filling |m_featureId| based on |m_osm_id| and probably seting |m_osm_id| to some invalid value. Not writing to mwm |m_osm_id| based on its name while visiting all Gate field to serialize it to mwm.

What do you think about that? The second variant looks better to me.

Reply:
Both approaches look good, but the second is less hacky.

Reply:
Yes. Let's implement the second approach. But this work is planed after I finish with filling |m_pedestrianFeatureIds| based on gate point. 

In this PR I'm adding |m_pedestrianFeatureIds| without any name and in next PR I'll fill it based on gate point.